File: canvas/js/sf3_conditionals.js
"use strict";

var sketch_3 = function(p) {

	var x = 100;
	var y = 100;
	var s = 50;
	var speed = 5;

	p.setup = function() {
		p.createCanvas(640, 200);
	}

	p.draw = function() {
		p.background(127);
		p.ellipse(x, y, s, s);
		x += speed;
	  
		if (x > p.width || x < 0) {
			speed *= -1;
		}
	}

}

var p5_3 = new p5(sketch_3, "canvas3");

File: p5py/sf3_conditionals.py
from p5 import *

x = 100
y = 100
s = 50
speed = 5

def setup():
    size(640, 360)

def draw():
    global x, y, s, speed
    background(127)
    ellipse(x, y, s, s)
    x += speed
    
    if (x > width or x < 0):
        speed *= -1

run()